contracts/UniswapV3Staker.sol

```diff
@@ -292,22 +295,32 @@ contract UniswapV3Staker is
             SafeMath.sub(incentives[incentiveId].totalRewardUnclaimed, reward)
         );
 
-        /* TODO: This will be fixed in https://github.com/omarish/uniswap-v3-staker/issues/38
-        so that collecting and unstaking are two separate functions */
-        try
-            IERC20Minimal(incentives[incentiveId].rewardToken).transfer(
-                params.to,
-                reward
-            )
-        returns (bool) {
-            // It didn't fail
-        } catch {
-            // It failed
-        }
+        rewards[incentives[incentiveId].rewardToken][msg.sender] = uint128(
+            SafeMath.add(rewards[incentives[incentiveId].rewardToken][msg.sender], reward)
+        );
 
         emit TokenUnstaked(params.tokenId);
     }
 
+    /// @inheritdoc IUniswapV3Staker
+    function claimReward(address rewardToken) external override {
+        uint128 reward = rewards[rewardToken][msg.sender];
+
+        require(reward != 0, 'NO_REWARDS_AVAILABLE');
+        
+        require(
+            IERC20Minimal(rewardToken).transfer(
+                msg.sender,
```

Comment:
in the original design the recipient is designated by a passed in param `to`. I think we can take that out of the `UnstakeParams` struct and pass it in here

Reply:
Cool, I can remove it from `UnstakeParams`. I'm not sure what you mean when you say "pass it in here".

Reply:
make it a parameter to this function, so you can send to `to` instead of `msg.sender`